Core/KSPManager.cs

```diff
@@ -200,67 +202,96 @@ public void CloneInstance (KSP existing_instance, string new_name, string new_pa
                 throw e;
             }
         }
-
         /// <summary>
         /// Create a new fake KSP instance
         /// </summary>
         /// <param name="new_name">The name for the new instance.</param>
         /// <param name="new_path">The loaction of the new instance.</param>
         /// <param name="version">The version of the new instance. Should have a build number.</param>
-        /// <param name="dlcVersion">The version of the DLC. Null if DLC should be faked.</param>
-        public void FakeInstance(string new_name, string new_path, KspVersion version, string dlcVersion = null)
+        /// <param name="dlcVersions">The versions of the DLCs. Null / empty strings if DLCs should not be faked.</param>
```

Comment:
I'm not a big fan of an array of strings as a parameter type, with the array indices meaningful and `null` padding values required in some cases. It'll require us to update this function every time a new DLC is released, and it's prone to array out of bounds errors and so forth. And the current parameter documentation doesn't explain how to use it.

What do you think about changing this parameter to a `Dictionary<IDlcDetector, KspVersion>` ? Then we could `foreach` through them to get the key value pairs, and use `InstallPath()` on the key to create the folders in a generic way. If you want to retain the game version validation, we could add a `bool AllowedOnBaseVersion(KspVersion)` method to `IDlcDetector` to do that generically as well. Then when a new DLC is released we would only have to update the calling code to add a new pair to the dictionary. Generating that parameter should be relatively simple:

```csharp
    new Dictionary<IDlcDetector, KspVersion>() {
        { new MakingHistoryDlcDetector(),  new KspVersion("1.1.0") },
        { new BreakingGroundDlcDetector(), new KspVersion("1.0.0") },
    }
```

Reply:
Sounds like a much better solution than mine. 👍

Reply:
I would add a new `private readonly KspVersion ReleaseGameVersion;` to the `StandardDlcDetectorBase` to save the KSP version there, and set it in the constructors like `IdentifierBaseName` and `DirectoryBaseName`.
I think it makes sense to have it in an obvious, code-accessible place, and not only buried somewhere inside `bool AllowedOnBaseVersion()`.
Does that seem reasonable to you?

Reply:
Definitely make whatever changes you think are appropriate for the `AllowedOnBaseVersion` concept, that was just meant to be very vague pseudocode. :+1:

Reply:
I'll try my best 😆

Reply:
One more thing:
Right now we output the DLC version as an `UnmanagedModuleVersion` in the detectors.
Should we 'upgrade' that to `KspVersion`? It allows better handling, and if I'm not mistaken it always matched the pattern, and probably will do so in the future too.

Reply:
I guess that sounds fine to me, but let's ask @dbent in case there are subtleties that I'm not noticing.

Reply:
The version object type is used as a sentinel value in a few different places to distinguish between modules CKAN installed itself and those that were autodetected. I don't particularly like that strategy myself but it's a legacy from how the original autodetect code was written.

If you return a `ModuleVersion` instead of an `UnmanagedModuleVersion` then everywhere that checks if the version object is an `UnmanagedModuleVersion` will have to be updated. IIRC, the changes were numerous and sensitive enough that it wasn't worth the effort at the time.

Reply:
Okay, thanks @dbent. So not something I want to do in this PR, if at all ^^
If this PR stays as is, we would be double-tracked, with all the instance faking stuff using `KspVersion` and the mod managing and other stuff using `UnmanagedModuleVersion`.
Well, it works, but I don't make use of it being `KspVersion`, so I probably should change it (back) to `UnmanagedModuleVersion` for the sake of consistency.

Reply:
On another thought, I could just leave it as is. The format checks should be done either way, and it should simplify a future change to `KspVersion` in the underlying module managing code.

Reply:
The new DLC creation implementation looks good now. :+1: